cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,563 @@
+        """
+
+        # Remove all the padding from the end
+        trimmed_collections = list()
+        max_arr_length = -1
+        dx = to_dlpack(tnsr)
+        embeddings_collecton = cp.fromDlpack(dx)
+        for embeddings in embeddings_collecton:
+            trimmed = cp.trim_zeros(embeddings, trim='b')
+            max_arr_length = max(max_arr_length, len(trimmed))
+            trimmed_collections.append(trimmed)
+
+        first_arr_stack = cp.pad(
+            trimmed_collections[0],
+            (0, max_arr_length-len(trimmed_collections[0])),
+            'constant')
+
+        # Add the required padding back
+        for a in range(1, len(trimmed_collections)):
+            padded = cp.pad(
+                trimmed_collections[a],
+                (0, max_arr_length-len(trimmed_collections[a])),
+                'constant')
+            first_arr_stack = cp.vstack([first_arr_stack, padded])
+            # Convert it back to a PyTorch tensor.
+        tx2 = from_dlpack(first_arr_stack.toDlpack())
+
+        # Taking care of case where we have only one sentence
+        # Then, we need to reshape to get the right dimensions
+        # since in the other cases cp.vstack handles that.
+
+        if len(tx2.shape) == 1:
+            dim = tx2.shape[0]
+            tx2 = torch.reshape(tx2, (1, dim))
+
+        return tx2
+
+    # Mean Pooling - Take attention mask into account for correct averaging
+    def mean_pooling(self, model_output, attention_mask):
+        """Function to implement mean pooling on top of the AutoModel
+        See: https://www.sbert.net/examples/applications/computing-embeddings/README.html#sentence-embeddings-with-transformers
+
+        Args:
+            model_output \
+                (transformers.BaseModelOutputWithPoolingAndCrossAttentions): BERT model
+            attention_mask (torch.Tensor): torch.Tensor representing attention
+            mask values
+
+        Returns:
+            [torch.Tensor]: correct averaging of attention mask
+        """
+        # First element of model_output contains all token embeddings
+        token_embeddings = model_output[0]
+        input_mask_expanded = attention_mask.\
+            unsqueeze(-1).\
+            expand(token_embeddings.size()).\
+            float()
+        sum_embeddings = torch.sum(token_embeddings * input_mask_expanded, 1)
+        sum_mask = torch.clamp(input_mask_expanded.sum(1), min=1e-9)
+        return sum_embeddings / sum_mask
+
+    def create_embeddings(self, sentences):
+        """Creates the sentence embeddings using SentenceTransformer
+
+        Args:
+            sentences (cudf.Series[str]): a cuDF Series of Input strings
+
+        Returns:
+            embeddings (cupy.ndarray): corresponding sentence
+            embeddings for the strings passed
+        """
+
+        cudf_tokenizer = SubwordTokenizer(
+            'voc_hash.txt',
+            do_lower_case=True
+        )
+
+        # Tokenize sentences
```

Comment:
Suggest adding `create_embeddings ` and related functions to a separate file to decrease the length of the file.